travis/getaddons.py

```diff
@@ -84,7 +84,138 @@ def get_addons(path):
         res = [os.path.join(path, x)
                for x in os.listdir(path)
                if is_addons(os.path.join(path, x))]
-    return res
+    deps = dict([(addon, get_dependencies(addon))
+                 for addon in res])
+    sorted_addons = get_sorted_addons_by_level(deps)
+    return sorted_addons
+
+
+def get_dependencies(path):
+    """Gets the dependencies of an addon reading the `oca_dependencies.txt`
+    file.
+
+    :param path: The addon path
+    :rtype: str
+
+    :returns: The dependency list
+    :rtype: list
+    """
+    deps = []
+    dep_files = ['oca_dependencies.txt']
+    for dep_file in dep_files:
+        dep_path = os.path.join(path, dep_file)
+        new_deps = []
+        try:
+            with open(dep_path) as dep:
+                for line in dep:
+                    line = line.strip()
+                    if not line or line.startswith('#'):
+                        continue
+                    new_deps.append(line.split()[0])
+        except IOError:
+            continue
+        deps = deps + new_deps
+    return deps
+
+
+def get_sorted_addons_by_level(addons):
+    """Sorts a list of addons according to its level, and followed
+    by its dependencies also sorted by level.
+
+    It first sorts the addons using `sort_addons_by_level` and then
+    it puts the dependencies after their parents. This to keep the addons
+    with higher dependency level first and then its addons also sorted by
+    level.
+
+    :param addons: The addons with their dependencies.
+    :type addons: dict
+
+    :returns: The addons sorted by level
+    :rtype: list
+
+    Example::
+
+        get_sorted_addons_by_level({
+            "e": [],
+            "r": ["x"],
+            "c": [],
+            "t": ["r", "c"],
+            "x": ["e"],
+        })
+
+    Returns::
+
+        ["t", "r", "c", "x", "e"]
+
+    """
+    sorted_addons = []
+    addons_by_level = sort_addons_by_level(addons)
+    addons_list = sorted(addons_by_level.keys(),
+                         key=lambda dep: -addons_by_level.get(dep))
+    for addon in addons_list:
+        if addon not in sorted_addons:
+            sorted_addons.append(addon)
+        for dep in addons.get(addon, []):
+            dep_addon = os.path.join(os.path.dirname(addon), dep)
+            if dep_addon not in sorted_addons:
+                sorted_addons.append(dep_addon)
```

Comment:
I can't use sets because I need to maintain the order of the elements added to the list, and using sets the order doesn't matter:

![image](https://user-images.githubusercontent.com/8657959/30281635-8f962e3a-96d8-11e7-9094-69e732ffacba.png)